packages/assets-controllers/src/token-prices-service/codefi-v2.ts

```diff
@@ -370,25 +389,52 @@ export class CodefiTokenPricesServiceV2
         // to keep track of them and make sure we return the original versions.
         const lowercasedTokenAddress =
           tokenAddress.toLowerCase() as Lowercase<Hex>;
-        const lowercasedCurrency =
-          currency.toLowerCase() as Lowercase<SupportedCurrency>;
 
-        const price =
-          pricesByCurrencyByTokenAddress[lowercasedTokenAddress]?.[
-            lowercasedCurrency
-          ];
+        const tokenData =
+          pricesByCurrencyByTokenAddress[lowercasedTokenAddress] || {};
 
-        if (!price) {
-          // console error instead of throwing to not interrupt the fetching of other tokens in case just one fails
-          console.error(
-            `Could not find price for "${tokenAddress}" in "${currency}"`,
-          );
-        }
+        const {
+          price,
+          pricePercentChange1d,
+          priceChange1d,
+          allTimeHigh,
+          allTimeLow,
+          circulatingSupply,
+          dilutedMarketCap,
+          high1d,
+          marketCap,
+          totalVolume,
+          low1d,
+          marketCapPercentChange1d,
+          pricePercentChange1h,
+          pricePercentChange7d,
+          pricePercentChange14d,
+          pricePercentChange30d,
+          pricePercentChange200d,
+          pricePercentChange1y,
+        } = tokenData;
 
         const tokenPrice: TokenPrice<Hex, SupportedCurrency> = {
           tokenAddress,
           value: price,
           currency,
+          pricePercentChange1d,
+          priceChange1d,
+          allTimeHigh,
+          allTimeLow,
+          circulatingSupply,
+          dilutedMarketCap,
+          high1d,
+          marketCap,
+          totalVolume,
+          low1d,
+          marketCapPercentChange1d,
+          pricePercentChange1h,
+          pricePercentChange7d,
+          pricePercentChange14d,
+          pricePercentChange30d,
+          pricePercentChange200d,
+          pricePercentChange1y,
```

Comment:
You can use the spread operator to avoid the verbose destructuring in `tokenData`
```
const tokenPrice: TokenPrice<Hex, SupportedCurrency> = {
  tokenAddress,
  ...tokenData,
  currency,
};
```